components/markdown/src/markdown.rs

```diff
@@ -139,7 +139,27 @@ fn fix_link(
                 resolved.permalink
             }
             Err(_) => {
-                return Err(anyhow!("Relative link {} not found.", link));
+                match context.config.link_checker.internal_level {
+                    config::LinkCheckerLevel::ErrorLevel => {
+                        return Err(anyhow!(
+                            "Dead relative link `{}` in {}",
+                            link,
+                            context.current_page_path.unwrap_or("unknown"),
+                        ))
+                    }
+                    config::LinkCheckerLevel::WarnLevel => {
+                        console::warn(
+                            format!(
+                                "{}Dead relative link `{}` in {}",
+                                config::LinkCheckerLevel::WarnLevel.log_prefix(),
+                                link,
+                                context.current_page_path.unwrap_or("unknown"),
+                            )
+                            .as_str(),
+                        );
+                        link.to_string() // TODO rendering the broken internal relative link may not be optimal, what href should be rendered to HTML when relative links are invalid but we're in warn mode?
```

Comment:
This needs a sanity check.  In the case of a broken relative link like `@/foobar`, previously the build would panic but if we're in `warn` mode, we need to render something into HTML.  Currently I'm writing the relative link directly into the href, so `<a href="@/foobar">` is what comes out the other end.  There's nothing we can really do to fix the link, but that seemed like the least surprising behavior.  Thoughts?

Reply:
I think your behaviour is fine if it's documented